src/backend/UserDb.js

```diff
@@ -11,17 +11,31 @@ class UserDb {
         return UserDb.instance;
     }
 
-    async userExists(un, pw) {
-        const doc = await this.programDb.cfind({ username: un, password: pw })
-            .exec();
+    async userExists(username, password = null) {
+        let doc = [];
+        if (password !== null) {
+            doc = await this.programDb.cfind({ username: username, password: password }).exec();
+        } else {
+            doc = await this.programDb.cfind({ username: username}).exec();
+        }
```

Comment:
You've now made it possible to log in without a password.

Reply:
We might want to update the UI to make it so people can't log in without a password tho. IDK if you should fix that since you are editing this function, or if it should be a seperate task